tests/test_testclient.py

```diff
@@ -87,10 +87,44 @@ def test_testclient_asgi3():
         await send({"type": "http.response.body", "body": b"Hello, world!"})
 
     client = TestClient(app)
-    response = client.get("/")
+    response = client.get("/one%2Ftwo")
     assert response.text == "Hello, world!"
 
 
+def test_testclient_scope():
+    app = Starlette()
+
+    @app.route("/one%2Ftwo")
+    def scope_route_explicit(request):
+        return JSONResponse({
+            "via_explicit_route": True,
+            "path": request.scope["path"],
+            "raw_path": str(request.scope["raw_path"]),
+        })
+
+    @app.route("/.*")
+    def scope_route_fallback(request):
+        return JSONResponse({
+            "path": request.scope["path"],
+            "raw_path": str(request.scope["raw_path"]),
+        })
+
+    client = TestClient(app)
+
+    # _ASGIAdapter unquotes "path", so this goes to the fallback route.
```

Comment:
How do you feel about not using "unquote" for the provided path?
(i.e. there is not really a reason for the test client to behave like a "bad webserver" that would unquote already)